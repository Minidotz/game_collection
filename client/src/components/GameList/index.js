import React from 'react';
import { List, ListItem, Avatar, ListItemText } from '@material-ui/core';
import { Image as ImgIcon} from '@material-ui/icons'
import {Link} from 'react-router-dom';

export default function GameList(props) {
    return (
        <List>
            {props.data.length > 0 ? (
                props.data.map(d => {
                    return (
                        <ListItem key={d._id} button component={Link} to={{pathname: "/games/" + d.guid, state: { title: d.title }}} >
                            {d.image ? (
                                <Avatar alt={d.title} src={d.image} />
                            ) : (
                                <Avatar><ImgIcon /></Avatar>
                            )}
                            <ListItemText primary={d.title} />
                        </ListItem>
                    )
                })
            ) : (
                    <ListItem>
                        <ListItemText primary={<em>No data</em>} />
                    </ListItem>
                )
            }
        </List>
    )
}