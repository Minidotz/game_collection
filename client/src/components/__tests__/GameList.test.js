import React from 'react';
import renderer from 'react-test-renderer';
import GameList from '../GameList';
import { MemoryRouter} from 'react-router-dom';
import { Avatar } from '@material-ui/core';
import { Image } from '@material-ui/icons';

it('renders', () => {
    let testData = [{
        _id: 1,
        guid: '678-9013',
        title: 'Test game',
        image: '/img/games/678-9013.jpg'
    }];
    const component = renderer.create(
        <MemoryRouter>
            <GameList data={testData} />
        </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
});

it('shows No data when there are no data props', () => {
    const component = renderer.create(
        <MemoryRouter>
            <GameList />
        </MemoryRouter>
    );
    const instance = component.root;
    expect(instance.findByType('em').children[0]).toBe('No data');
});

it("shows an Avatar with a generic icon when there's no image in game data", () => {
    let testData = [{
        _id: 1,
        guid: '678-9013',
        title: 'Test game'
    }]
    const component = renderer.create(
        <MemoryRouter>
            <GameList data={testData} />
        </MemoryRouter>
    );
    const instance = component.root;
    expect(instance.findByType(Avatar).findByType(Image)).toBeTruthy();
});