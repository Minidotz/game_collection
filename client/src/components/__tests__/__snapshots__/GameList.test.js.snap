// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/games/678-9013"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root"
      >
        <img
          alt="Test game"
          className="MuiAvatar-img"
          src="/img/games/678-9013.jpg"
        />
      </div>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Test game
      </span>
    </div>
  </a>
</ul>
`;
